### DISCUSSION TITLE
[web] How can I open the sqlite database in memory instead of using LocalStorage

### DISCUSSION THREAD

**Discussion by thachnb85** [simolus3/drift#1082] [Q&A] [answered]
Hi,
I am going to use moor web but I found the limitation of 5MB for LocalStorage.
Sqlite supports in-memory option: https://sqlite.org/inmemorydb.html

Does it support by moor? or where can modify to get it work with in-memory instead of LocalStorage.

Thank you.

**Comment by simolus3** [comment, marked as answer]
That's an interesting question! You could write a custom `MoorWebStorage` that just stores data in memory:

```dart
class InMemoryWebStorage implements MoorWebStorage {
  Uint8List? _storedData;

  @override
  Future<void> close() => Future.value();

  @override
  Future<void> open() => Future.value();

  @override
  Future<Uint8List?> restore() => Future.value(_storedData);

  @override
  Future<void> store(Uint8List data) {
    _storedData = data;
    return Future.value();
  }
}
```

You could then use a `WebDatabase.withStorage(InMemoryWebStorage())`.

**Reply by thachnb85** [reply]
Thank you @simolus3 I've tested and the demo app on the web works properly, you may consider adding to official release.
Cheers,
